Sources/ArcGISToolkit/Components/FormView/FormInputs/FormInputModel.swift

```diff
@@ -49,33 +49,33 @@ import SwiftUI
         formattedValue = element.formattedValue
         
         // Kick off tasks to monitor required, editable and value.
-        observationTask = Task {
+        observationTask = Task.detached { [unowned self] in
             await withTaskGroup(of: Void.self) { group in
                 // Observe isRequired changes.
-                group.addTask { [unowned self] in
-                    for await isRequired in await element.$isRequired {
+                group.addTask {
+                    for await isRequired in await self.element.$isRequired {
                         guard !Task.isCancelled else { return }
                         await MainActor.run {
                             self.isRequired = isRequired
```

Comment:
I'm able to reproduce the crash here again. Unfortunately when I go to record what I'm doing it doesn't crash.

<img width="580" alt="Screenshot 2023-12-19 at 14 11 11" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/ddba372b-33eb-4d7e-8dd0-7a4a7ace5f6a">

> Fatal error: Attempted to read an unowned reference but object 0x60000ac3c380 was already deallocated

Reply:
@philium I've tried to use [weak self] for this, but can't get past the compiler errors. I usually get _Reference to captured var 'self' in concurrently-executing code_, but sometimes it's _Value of optional type 'FormInputModel?' must be unwrapped to refer to member 'element' of wrapped base type 'FormInputModel'_, which I can fix by using `self?.`, but then I get the _For-in loop requires 'AsyncStream<Bool>?' to conform to 'AsyncSequence'; did you mean to unwrap optional?_ error.

Reply:
Does the guard possibly need to be moved down into the closure sent to the main actor? It fixes the crash for me.

E.g.
```swift
for await isRequired in await self.element.$isRequired {
    await MainActor.run {
        guard !Task.isCancelled else { return }
        self.isRequired = isRequired
    }
}
```

Reply:
As I started looking into this, I couldn't help but get the feeling that the `task` modifier might be a better approach to using a model that creates and manages its own task with child tasks. I see why the model approach was used—the code needed to be re-usable because it is used by multiple views. But there's more than one way to make code re-usable. The model-based approach is problematic not only because of needing to manage the task but also because it needs to be initialized with an element. Due to the nature of `StateObject`, the model is only created once per instance of the view on-screen. If the element changes while the view is displayed, the model won't get re-created and the view will be in a bad state (since the model would have a stale element).

Instead of using a model, consider adding convenience modifiers for observing the element:

```swift
extension View {
    func onChangeOfIsEditable(
        of element: FieldFormElement,
        action: @escaping (_ newIsEditable: Bool) -> Void
    ) -> some View {
        return self
            .task(id: ObjectIdentifier(element)) {
                for await isEditable in element.$isEditable {
                    action(isEditable)
                }
            }
    }
    
    func onChangeOfIsRequired(
        of element: FieldFormElement,
        action: @escaping (_ newIsRequired: Bool) -> Void
    ) -> some View {
        return self
            .task(id: ObjectIdentifier(element)) {
                for await isRequired in element.$isRequired {
                    action(isRequired)
                }
            }
    }
    
    func onChangeOfValue(
        of element: FieldFormElement,
        action: @escaping (_ newValue: Any?, _ newFormattedValue: String) -> Void
    ) -> some View {
        return self
            .task(id: ObjectIdentifier(element)) {
                for await value in element.$value {
                    action(value, element.formattedValue)
                }
            }
    }
}
```

If you really want to, you could combine those into a single modifier, although I'm not sure that really is that much of an improvement—the first closure not having a label at the point of use looks really odd:

```swift
func observe(
    _ element: FieldFormElement,
    isRequiredChanged: @escaping (_ newIsRequired: Bool) -> Void,
    isEditableChanged: @escaping (_ newIsEditable: Bool) -> Void,
    valueChanged: @escaping (_ newValue: Any?, _ newFormattedValue: String) -> Void
) -> some View {
    return self
        .onChangeOfIsEditable(of: element, action: isEditableChanged)
        .onChangeOfIsRequired(of: element, action: isRequiredChanged)
        .onChangeOfValue(of: element, action: valueChanged)
}
```

In the views that use the model, replace the state object model property with four state properties. Perhaps even just three, depending on how `formattedValue` is used. It might be possible to do all that is needed related to `formattedValue` in the action closure for `onChangeOfValue` so that it doesn't need to be stored as state. Then use the modifiers (or single modifier) to update those state properties and use them in the view's body where needed. No more model. No more manual task management.

What do you think?

Reply:
I think that sounds good; It's a bit more code, but it probably will work better for the reasons you mention.  Thanks!